code/game/objects/items/devices/flashlight.dm

```diff
@@ -100,37 +100,40 @@
 	else
 		..()
 
-/obj/item/device/flashlight/attack(mob/living/M as mob, mob/living/user as mob)
+/obj/item/device/flashlight/attack(mob/living/carbon/human/M as mob, mob/living/user as mob)
 	add_fingerprint(user)
 	if(on && user.zone_selected == "eyes")
 
 		if((user.getBrainLoss() >= 60) && prob(50)) //too dumb to use flashlight properly
 			return ..() //just hit them in the head
 
-		if((!ishuman(user) || SSticker) && SSticker.mode.name != "monkey") //don't have dexterity
+		if (!(istype(user, /mob/living/carbon/human) || SSticker) && SSticker.mode.name != "monkey") //don't have dexterity
```

Comment:
why did you replace this

Reply:
That's the exact code line that wasn't allowing the flashing to work, so I simply replaced it with a more modern counterpart, and now works without any problems at all

Reply:
The older line is possibly outdated

Reply:
The ishuman macro will 100% work

Reply:
> The ishuman macro will 100% work

I have tested it myself, and that's the cause of the malfunction

Reply:
resolved